### PR TITLE
core: change output pattern of translations

### PR DESCRIPTION
**Pull request by alvaropag**
This should help fix zotonic 1377 (https://github.com/zotonic/zotonic/issues/1377).
Some more changes are needed on zotonic/mod_translation.

### CODE REVIEW

File: src/template_compiler.erl [hunk 1]
```diff
@@ -301,7 +301,8 @@ translations(Filename) ->
             end;
         {error, _} = Error ->
             Error
-    end.
+    end,
+    {Filename, Translations}.
```
Comment (arjan) [inline src/template_compiler.erl]: Why would you return the filename? It is given as an input to the function, so returning it as well seems unnecessary.

Comment (mworrell) [inline src/template_compiler.erl]: I agree, keep it simple :)

**Comment by mworrell** [comment]
If I understand correctly, then the translation system wants to have a list of all translated terms with their source position?

(btw this also changed the `-spec` of the `translations/1` function)

**Comment by alvaropag** [comment]
Yes, I'm following the pattern returned by mod_translation/support/translation_scan/scan_file. 
And, as I understand it was also used by the erlydtl library.
Sorry, I should have corrected the spec too.

So, what pattern should we return? Just the strings, or a tuple with more information.

**Comment by mworrell** [comment]
I am for: `{ok, list()} | {error, term()}`

**Comment by alvaropag** [comment]
I can aggregate the filename on mod_translation to generate the PO file, so {ok, list()} is good too.
I'll keep the tuple structure as this pull request (with line number info).

**Pull request closed without merging**